app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    def set_spouse(self, spouse_name: str) -> None:
```

Comment:
I think, you can implement task without additional methods. But if it was more convenient for you to realize or understand the task, then let it remain as it is.